deepdrivemd/aggregation/stream/aggregator.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+from deepdrivemd.aggregation.stream.config import StreamAggregation
+from deepdrivemd.utils import Timer, timer, parse_args, intarray2hash
+import time
+import os
+import sys
+import adios2
+import math
+import queue
+import subprocess
+import itertools
+from deepdrivemd.data.stream.adios_utils import AdiosStreamStepRW
+from typing import List, Dict, Tuple
+from pathlib import Path
+
+
+def find_input(cfg: StreamAggregation) -> List[Path]:
+    """Find adios streams to which simulations write.
+
+    Parameters
+    ----------
+    cfg : StreamAggregation
+
+    Returns
+    -------
+    List[Path]
+           a list of sst files associated with simulations
+    """
+    while True:
+        bpfiles = list(map(str, list(cfg.experiment_directory.glob("*/*/*/md.bp*"))))
+        if len(bpfiles) == cfg.n_sim:
+            break
+        print("In find_input: waiting for input")
+        time.sleep(cfg.sleeptime_bpfiles)
+
+    bpfiles.sort()
+    return bpfiles
+
+
+def connect_to_input(
+    cfg: StreamAggregation, bpfiles: List[Path]
+) -> Dict[int, Tuple[adios2.adios2.ADIOS, adios2.adios2.IO, adios2.adios2.Engine]]:
+    """Open adios streams for reading.
+
+    Parameters
+    ----------
+    cfg : StreamAggregation
+    bpfiles : List[Path]
+
+    Returns
+    -------
+    Dict[int, Tuple[adios2.adios2.ADIOS, adios2.adios2.IO, adios2.adios2.Engine]]
+           key - simulation task id, value - tuple of the corresponding adios objects.
+
+    """
+    connections = {}
+    bp_slice = math.ceil(cfg.n_sim / cfg.num_tasks)
+    print("bp_slice = ", bp_slice)
+
+    for i, bp in enumerate(bpfiles):
+        bp = bp.replace(".sst", "")
+        dir = os.path.dirname(bp)
```

Comment:
`dir` is a builtin Python variable, can we change the name of this variable?

Reply:
Replaced with sim_dir